### PR TITLE
Classnotes

### PR DESCRIPTION
**Pull request by amarisch**
@jontey

### CODE REVIEW

File: ap/ap/urls.py [hunk 1]
```diff
@@ -22,13 +22,46 @@
...
 from web_access.forms import WebAccessRequestGuestCreateForm as form
+from classnotes.views import ClassNoteViewSet
 
 from rest_framework_nested import routers
 from rest_framework_bulk.routes import BulkRouter
 
 
 admin.autodiscover()
 
+<<<<<<< HEAD
```
Comment (jontey) [inline ap/ap/urls.py]: ===========

File: ap/ap/urls.py [hunk 2]
```diff
@@ -59,6 +92,7 @@
   url(r'^admin/', include("massadmin.urls")),
   url(r'^static/(?P<path>.*)$', 'django.views.static.serve', {'document_root': settings.STATIC_ROOT}),
 ] + static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
+>>>>>>> 15d390e80b111afac5731b997e2c7e332e6eda05
```
Comment (jontey) [inline ap/ap/urls.py]: merge leftovers

File: ap/ap/urls.py [hunk 3]
```diff
@@ -81,13 +115,17 @@
 router.register(r'seats', SeatViewSet)
 router.register(r'partials', PartialViewSet)
 router.register(r'terms', TermViewSet)
+<<<<<<< HEAD
```
Comment (jontey) [inline ap/ap/urls.py]: same here

File: ap/classnotes/static/classnotes/js/ajax-api-binding.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * Binds ajax api calls to buttons for lifestudy pages
+ *
+ */
+
+$(window).ready(function(e) {
+  // Bind click event to all delete buttons to ajax delete summaries
+  $('.delete').on('click', function(ev) {
+    var btn = $(this);
+    if (!window.confirm('Are you sure you want to delete this life study summary?')) {
```
Comment (jontey) [inline ap/classnotes/static/classnotes/js/ajax-api-binding.js]: classnotes

File: ap/classnotes/static/classnotes/js/ajax-api-binding.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+/**
```
Comment (jontey) [inline ap/classnotes/static/classnotes/js/ajax-api-binding.js]: is this page necessary?

File: ap/classnotes/templates/classnotes/generic_list.html [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+{% else %}
+  <h4><i>{% block no-notes %}{% endblock %}</i></h4>
+{% endif %}
+
+{% block classnotes_content %}{% endblock %}
+
+{% endblock %}
+
+{% block scripts %}
+<script src="{% static "jquery/js/jquery-ui.js" %}"></script>
```
Comment (jontey) [inline ap/classnotes/templates/classnotes/generic_list.html]: i think this is already in npm bundle

File: ap/classnotes/urls.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+urlpatterns = patterns('',
+	url(r'^$', views.ClassnotesListView.as_view(), name='classnotes_list'),
+	url(r'^classnotes-assign$', views.ClassnotesAssignView.as_view(),
+        name='classnotes_assign'),
+	url(r'^classnotes-report$', views.ClassnotesReportView.as_view(),
+        name='classnotes_report'),
+	url(r'^(?P<pk>\d+)/detail-classnotes$', views.ClassnotesUpdateView.as_view(),
+		name='classnotes_detail'),
+    url(r'^(?P<pk>\d+)/approve-classnotes$', views.ClassnotesApproveView.as_view(),
```
Comment (jontey) [inline ap/classnotes/urls.py]: tab is off

File: ap/classnotes/utils.py [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+from attendance.utils import Period
+from leaveslips.models import IndividualSlip, GroupSlip
+from terms.models import Term
+from accounts.models import Trainee
+from classes.models import Class
+
+
+def assign_classnotes():
+
+	trainee = Trainee.objects.get(pk=7)
```
Comment (jontey) [inline ap/classnotes/utils.py]: Is this for testing?

File: ap/classnotes/utils.py [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+	rolls = rolls.extra(order_by = ['date'])
+	for roll in rolls:
+			classname = roll.event.name
+			number_classnotes = calculate_number_classnotes(trainee, roll)
+			leavesliplist = get_leaveslip(trainee, roll)
+			if leavesliplist:
+				for leaveslip in leavesliplist:
+					# Special: Wedding, Graduation, Funeral, etc.
+					if leaveslip.type == 'FWSHP' or leaveslip.type == 'FUNRL' or \
+        				leaveslip.type == 'INTVW' or leaveslip.type == 'GRAD' or \
```
Comment (jontey) [inline ap/classnotes/utils.py]: maybe we can shorten this to `leaveslip.type in ['INTVW', 'GRAD', 'WED']`

File: ap/classnotes/utils.py [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+        				leaveslip.type == 'WED':
+						generate_classnotes(trainee, roll, 'S')
+
+					# Regular: Sickness, Unexcused, Others, Fellowship, Notif only
+					if leaveslip.type == 'OTHER' or leaveslip.type == 'SICK' or leaveslip.type == 'FWSHP' \
```
Comment (jontey) [inline ap/classnotes/utils.py]: same for the rest of the if statements

File: ap/templates/base.html [hunk 11]
```diff
@@ -83,6 +83,62 @@
 
     <div class="container-fluid">
       <div class="row row-offcanvas row-offcanvas-left">
+<<<<<<< HEAD
```
Comment (jontey) [inline ap/templates/base.html]: Merge leftovers

File: ap/classnotes/utils.py [hunk 12]
```diff
@@ -11,12 +11,9 @@
 
 def assign_classnotes():
 
-	trainee = Trainee.objects.get(pk=7)
-	update_classnotes_list(trainee)
-	assign_individual_classnotes(trainee)
-	# for trainee in Trainee.objects.filter(is_active=True).all():
-	#	update_classnotes_list(trainee)
-	# 	assign_individual_classnotes(trainee)
+	for trainee in Trainee.objects.filter(is_active=True).all():
```
Comment (jontey) [inline ap/classnotes/utils.py]: Trainee.objects.all() is sufficient. the Trainee Manager object handles the is_active by default

File: ap/accounts/management/commands/populate.py [hunk 13]
```diff
@@ -5,9 +5,9 @@ class Command(BaseCommand):
   def handle(self, *args, **options):
     call_command('populate_terms')
     call_command('populate_events')
-    call_command('populate_schedules')
     call_command('populate_trainees')
     call_command('populate_tas')
     call_command('populate_rolls')
     call_command('populate_services')
+    call_command('populate_schedules')
```
Comment (billyeh) [inline ap/accounts/management/commands/populate.py]: 👍

**Comment by jontey** [comment]
Looks good for initial commit. Merging!
